src/ModelContextProtocol/Server/AIFunctionMcpServerTool.cs

```diff
@@ -338,12 +349,111 @@ public override async ValueTask<CallToolResponse> InvokeAsync(
                 {
                     Text = JsonSerializer.Serialize(result, AIFunction.JsonSerializerOptions.GetTypeInfo(typeof(object))),
                     Type = "text"
-                }]
+                }],
+                StructuredContent = structuredContent,
             },
         };
     }
 
-    private static CallToolResponse ConvertAIContentEnumerableToCallToolResponse(IEnumerable<AIContent> contentItems)
+    private static JsonElement? CreateOutputSchema(AIFunction function, McpServerToolCreateOptions? toolCreateOptions, out bool structuredOutputRequiresWrapping)
+    {
+        // TODO replace with https://github.com/dotnet/extensions/pull/6447 once merged.
+
+        structuredOutputRequiresWrapping = false;
```

Comment:
The AIFunction output won't contain this bool, and so at that point we'll need to determine this from the JsonElement returned from the AIFunction. Do you want to structure this similarly, i.e. have a RequiresWrapping(JsonElement) method that is called separately from CreateOutputSchema?

Reply:
Can we just do that now then?